From: Jeff Epler <jepler@unpythonic.net>
Subject: `zpool status` incorrectly names raidz vdevs
Forwarded: no

A minimal version of Jeff Epler's original patch, based on a
similar fix in ZFS-on-Linux.

In the RAID-Z case, path = buf already, so snprintf src and dst
buffers would be the same, which doesn't work with GNU libc.  Use
another buffer instead.

--- a/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
+++ b/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
@@ -3404,7 +3404,7 @@
 {
 	char *path, *devid;
 	uint64_t value;
-	char buf[64];
+	char buf[64], buf2[64];
 	vdev_stat_t *vs;
 	uint_t vsc;
 	int have_stats;
@@ -3516,9 +3516,9 @@
 
 			verify(nvlist_lookup_uint64(nv, ZPOOL_CONFIG_ID,
 			    &id) == 0);
-			(void) snprintf(buf, sizeof (buf), "%s-%llu", path,
+			(void) snprintf(buf2, sizeof (buf2), "%s-%llu", path,
 			    (u_longlong_t)id);
-			path = buf;
+			path = buf2;
 		}
 	}
 
